fraim/actions/github.py

```diff
@@ -0,0 +1,153 @@
+                # GitHub personal access tokens start with 'ghp_', 'gho_', or 'ghu_'
+                # or could be classic tokens that start with letters/numbers
+                if password and (password.startswith(("ghp_", "gho_", "ghu_")) or len(password) >= 20):
+                    logger.info("GitHub token found via git credential helper")
+                    return password
+    except subprocess.CalledProcessError as e:
+        logger.debug(f"Git credential helper failed: {e}")
+
+    logger.warning("No GitHub token found from any source")
+    return None
+
+
+def add_reviewer(pr_url: str, description: str, user_or_group: str) -> None:
+    """
+    Notifies a GitHub user or group by adding them as reviewers and leaving a comment on a PR.
+
+    Args:
+        pr_url: The full URL to the GitHub PR
+        description: The description to add as a comment
+        user_or_group: The GitHub user or group to notify (format: @org/team-name for groups, username for users)
+
+    Raises:
+        ValueError: If the PR URL is invalid or the user_or_group format is incorrect
+        RuntimeError: If GitHub token is not set or API calls fail
+    """
+    logger.info(f"Notifying GitHub user_or_group {user_or_group} for PR: {pr_url}")
+
+    # Parse PR URL to get owner, repo, and PR number
+    logger.debug(f"Parsing PR URL: {pr_url}")
+    try:
+        path_parts = urlparse(pr_url).path.strip("/").split("/")
+        if len(path_parts) < 4 or path_parts[-2] != "pull":
+            raise ValueError
+        owner, repo, _, pr_number = path_parts
+        logger.debug(f"Parsed PR details - Owner: {owner}, Repo: {repo}, PR: {pr_number}")
+    except (ValueError, IndexError):
+        logger.error(f"Failed to parse PR URL: {pr_url}")
+        raise ValueError(f"Invalid PR URL format: {pr_url}")
+
+    # Get GitHub token from multiple sources
+    logger.debug("Getting GitHub authentication token")
+    github_token = _get_github_token()
+    if not github_token:
+        logger.error("GitHub authentication failed - no token found")
+        raise RuntimeError(
+            "GitHub authentication required. Please ensure one of the following:\n"
+            "1. Set GITHUB_TOKEN environment variable\n"
+            "2. Configure git credentials for github.com\n"
+            "3. Login with GitHub CLI (`gh auth login`)"
+        )
+
+    # Initialize GitHub client
+    logger.debug("Initializing GitHub client")
+    gh = Github(github_token)
+
+    try:
+        # Get repository and PR
+        logger.debug(f"Getting repository: {owner}/{repo}")
+        repository = gh.get_repo(f"{owner}/{repo}")
+        logger.debug(f"Getting pull request #{pr_number}")
+        pull_request = repository.get_pull(int(pr_number))
+    except Exception as e:
+        logger.error(f"Failed to get repository or pull request: {str(e)}")
+        raise RuntimeError(f"Failed to get repository or pull request: {str(e)}")
```

Comment:
```suggestion
        raise RuntimeError(f"Failed to get repository or pull request: {str(e)}") from e
```
so we don't lose the cause.